exec/java-exec/src/main/resources/rest/profile/list.ftl

```diff
@@ -37,7 +37,42 @@
       <strong>No running queries.</strong>
     </div>
   </#if>
-  <h3>Completed Queries</h3>
+  <table width="100%">
+    <script type="text/javascript" language="javascript">
+    //Validate that the fetch number is valid
+    function checkMaxFetch() {
+      var maxFetch = document.forms["profileFetch"]["max"].value;
+      console.log("maxFetch: " + maxFetch);
+      if (isNaN(maxFetch) || (maxFetch < 1) || (maxFetch > 100000) ) {
+        alert("Invalid Entry: " + maxFetch + "\n" +
+               Please enter a valid number of profiles to fetch (1 to 100000) ");
+        return false;
+      }
+      return true;
+    }
+    </script>
+    <tr>
+      <td><h3>Completed Queries</h3></td>
+      <td align="right">
+        <form name="profileFetch" action="/profiles" onsubmit="return checkMaxFetch();" method="get"><span title="Max number of profiles to list">Showing <b>${model.getFinishedQueries()?size}</b> profiles. Max: </span>
+        <input id="fetchMax" type="text" size="7" name="max" value="">
```

Comment:
Here we want to show the current fetch max setting. This is either:

* The default number, if the user has nor provided a value.
* The last submitted value.

Let's say the default is 100. Then we'd want to first see:

Max: [  100] [Fetch]

The user decides to view just 20 by changing the number and clicking Fetch. Now we'd want to see:

Max: [   20] [Fetch]

Also, "Fetch" is a bit awkward. Maybe "Refresh" would be better.

Reply:
Done. 
1. The keyword is now 'Refresh'.
2. The default fetch is populated into the textbox. Any subsequent changes are retained, so the user does not have to re-enter a value to trigger a reload. Only click on Refresh.